tools/ci_connector_ops/ci_connector_ops/pipelines/utils.py

```diff
@@ -155,3 +157,12 @@ def get_modified_connectors(modified_files: Set[str]) -> Set[Connector]:
         if file_path.startswith(SOURCE_CONNECTOR_PATH_PREFIX) or file_path.startswith(DESTINATION_CONNECTOR_PATH_PREFIX):
             modified_connectors.append(Connector(get_connector_name_from_path(file_path)))
     return set(modified_connectors)
+
+
+async def execute_steps_concurrently(steps: List[Callable], concurrency: int = 5):
```

Comment:
```suggestion
async def execute_steps_concurrently(steps: List[Step], concurrency: int = 5) -> List[StepResults]:
```

Reply:
I can rename this function because in this case the implementation doesn't depend on the arguments being a Step

Reply:
and add a generic

Reply:
True, feel free to rename to `execute_concurrently`.
This functions assumes the callable has no arguments. 
Feel free to add `*args, **kwargs` to the signature for improved flexibility.